python/dgl/heterograph.py

```diff
@@ -4651,6 +4651,30 @@ def to_format(self, restrict_format):
                               self._node_frames,
                               self._edge_frames)
 
+    # TODO: Formats should not be specified, just saving all the materialized formats
+    def shared_memory(self, name, formats=('coo', 'csr', 'csc')):
+        """Return a copy of this graph in shared memory (without node feature or edge feature)
+
+        Parameters
+        ----------
+        name : str
+            The name of the shared memory.
+        formats : list of str (optional)
+            Desired formats to be materialized.
+
+        Returns
+        -------
+        HeteroGraph
+            The graph index in shared memory
+        """
+        assert len(name) > 0, "The name of shared memory cannot be empty"
+        assert len(formats) > 0
+        for fmt in formats:
+            assert fmt in ("coo", "csr", "csc")
+        gidx = self._graph.shared_memory(name, self.ntypes, self.etypes, formats)
+        return DGLHeteroGraph(gidx, self.ntypes, self.etypes)
```

Comment:
so this operation only moves the graph index to shared memory and the returned DGLHeterograph doesn't contain node/edge data. We need to clearly define the semantics in the docstring.